### DISCUSSION TITLE
How can I use GitHub copilot with pluto notebooks ?

### DISCUSSION THREAD

**Discussion by Snimm** [fonsp/Pluto.jl#2408] [Questions about using Pluto] [answered]
How can I use GitHub copilot with Pluto notebooks?

**Comment by pankgeorg** [comment]
You can open the notebook in VSCode and use it from there :)

**Comment by Snimm** [comment]
@pankgeorg  I am using [Pluto.jl (alpha preview) ](https://marketplace.visualstudio.com/items?itemName=JuliaComputing.pluto-vscode)extension by JuliaHub Inc to run Pluto in VScode. However, I am unable to use Github copilot as it gives no suggestion.   Github copilot does give suggestion if I am using vanilla Julia in VScode.

**Comment by pankgeorg** [comment, marked as answer]
No I meant the actual VSCode editor; Pluto.jl files are also text files that you can edit in any text editor. We don't have integration with CoPilot, in part because it is not free.

**Comment by Snimm** [comment]
So there is no way to take advantage of both Pluto and CoPilot at the same time?

**Comment by pankgeorg** [comment]
No

**Comment by TheFibonacciEffect** [comment]
After I edited the file using VS code, is there any way to "reload" the file from the disk? So that I see the changes appear in pluto?

**Reply by holomorphism** [reply]
@TheFibonacciEffect If you start Pluto with the `auto_reload_from_file` option enabled, Pluto will automatically update its state when the notebook file is rewritten in an external editor:

```julia
Pluto.run(auto_reload_from_file=true)
```

See this pull request for details: #1029

There does not seem to be a way to manually reload a file in Pluto (other than closing the notebook and reopening it again).

**Reply by TheFibonacciEffect** [reply]
Thanks a lot : )
